resolwe/flow/utils/docs/autoprocess.py

```diff
@@ -0,0 +1,395 @@
+              with only this process is returned.
+            * If none of this options, then raise a ValueError
+
+        :param list content: content of the directive
+        :param set categories: set of all categories
+        :param dict processes_dict: dict of all source_base_url: processes pairs
+        :rtype: list
+        """
+        if not content:
+            return processes_dict
+        elif len(content) > 1:
+            raise NotImplementedError("Only one argument can be passes to autoprocess.")
+        content = content[0]
+        # Make a filtered dict of all process that have name == content
+        querry_for_slug = dict(pair for pair in processes_dict.items() if pair[1]['slug'] == content)
+        if len(querry_for_slug) == 1:
+            return querry_for_slug
+
+        elif content in categories:
+            filtered_dict = dict(
+                pair for pair in processes_dict.items()
+                if pair[1].get('category', 'uncategorized') == content)
+            return filtered_dict
+
+        raise ValueError("Invalid argument given to autoprocess directive.")
+
+    def get_process_definition_start(self, fname, slug):
+        """
+        Find the starting line of process definition - where slug is defined
+
+        :param path/str fname: Filename with processes
+        :param string slug: slug of the process
+        :return: line where the process definiton starts
+        :rtype: int
+        """
+        with open(fname) as file_:
+            for i, line in enumerate(file_):
+                if re.search(r'slug:\s*{}'.format(slug), line):
+                    return i + 1
+        # In case starting line is not found just return first line
+        return 1
+
+    def get_all_processes(self, process_dir, source_base_url):
+        """
+        Return a dictionary of all processes in `process_dir`
+
+        Dictionary key is an URL's to process source code and value is
+        process as it is loaded from yaml file.
+
+        :param path/str process_dir: Parent directory of all process files
+        :param url/string source_base_url: Base of the url to source-code repository
+        :rtype: dict
+        """
+        all_process_files = []
+        process_file_extensions = ['*.yaml', '*.yml']
+        for root, _, filenames in os.walk(process_dir):
+            for extension in process_file_extensions:
+                for filename in fnmatch.filter(filenames, extension):
+                    all_process_files.append(os.path.join(root, filename))
+
+        def read_yaml_file(fname):
+            """Read yaml file"""
+            with open(fname) as file_:
+                return yaml.load(file_)
+
+        all_processes = {}
+        for yaml_file in all_process_files:
+            processes_in_file = read_yaml_file(yaml_file)
+            for process in processes_in_file:
+
+                # This section finds the line in file where the
```

Comment:
This comment/description is probably redundant since `get_process_definition_start()` is already nicely documented?